tardis/simulation/base.py

```diff
@@ -401,6 +406,137 @@ def to_hdf(self, model, path_or_buf, path='', plasma_properties=None):
+        z_ct_outer      = z_ct[:, ps > R_min_rel]
+        n_shell_p_outer = (num_shell+1) - np.isnan(z_crossings[:, ps > R_min_rel]).sum(axis=0)
+
+        ## p < Rmin
+        ps_inner        = ps[ ps <= R_min_rel]
+        z_ct_inner      = z_ct[:, ps <= R_min_rel]
+
+        # I will traverse from largest shell in, but
+        # elsewhere shell structure is from smallest and out,
+        # so this is for reversing it
+        shell_nr = np.arange(0,num_shell,dtype="int")[::-1]
+
+ 
+        nus = np.linspace(4000, 7000, 20)*u.angstrom
+        nus = co.c.cgs / nus.cgs
+      
+#        nus = np. array([  7.28882474e+12,   4.72768145e+13,   8.72648043e+13, 1.27252794e+14,   1.67240784e+14,   2.07228774e+14, 2.47216763e+14,   2.87204753e+14,   3.27192743e+14,          3.67180733e+14,   4.07168722e+14,   4.47156712e+14, 4.87144702e+14,   5.27132692e+14,   5.67120682e+14, 6.07108671e+14,   6.47096661e+14,   6.87084651e+14,  7.27072641e+14,   7.67060630e+14,   8.07048620e+14,          8.47036610e+14,   8.87024600e+14,   9.27012590e+14, 9.67000579e+14,   1.00698857e+15,   1.04697656e+15, 1.08696455e+15,   1.12695254e+15,   1.16694053e+15,  1.20692852e+15,   1.24691651e+15,   1.28690450e+15,          1.32689249e+15,   1.36688048e+15,   1.40686847e+15, 1.44685646e+15,   1.48684445e+15,   1.52683244e+15, 1.56682043e+15,   1.60680842e+15,   1.64679641e+15,  1.68678440e+15,   1.72677238e+15,   1.76676037e+15,          1.80674836e+15,   1.84673635e+15,   1.88672434e+15, 1.92671233e+15,   1.96670032e+15,   2.00668831e+15, 2.04667630e+15,   2.08666429e+15,   2.12665228e+15,  2.16664027e+15,   2.20662826e+15,   2.24661625e+15,          2.28660424e+15,   2.32659223e+15,   2.36658022e+15, 2.40656821e+15,   2.44655620e+15,   2.48654419e+15, 2.52653218e+15,   2.56652017e+15,   2.60650816e+15,  2.64649615e+15,   2.68648414e+15,   2.72647213e+15,          2.76646012e+15,   2.80644811e+15,   2.84643610e+15, 2.88642409e+15,   2.92641208e+15,   2.96640007e+15, 3.00638806e+15,   3.04637605e+15,   3.08636404e+15,  3.12635203e+15,   3.16634002e+15,   3.20632801e+15,          3.24631600e+15,   3.28630399e+15,   3.32629198e+15, 3.36627997e+15,   3.40626796e+15,   3.44625595e+15, 3.48624393e+15,   3.52623192e+15,   3.56621991e+15,  3.60620790e+15,   3.64619589e+15,   3.68618388e+15,          3.72617187e+15,   3.76615986e+15,   3.80614785e+15, 3.84613584e+15,   3.88612383e+15,   3.92611182e+15, 3.96609981e+15,   4.00608780e+15])/u.s
+
+        #Just aliasing for cleaner expressions later
+        line_nu  = model.plasma.lines.nu
+        taus     = model.plasma.tau_sobolevs
+        att_S_ul = self.runner.att_S_ul
+        temps    = model.plasma.t_rad
+
+        dtau = 1 # Just to remember it 
+
+        L_nu  = np.zeros(nus.shape)
+
+        import pdb
+        for nu_idx,nu in enumerate(nus.value):
+            I_outer = np.zeros(ps_outer.shape)
+            for p_idx,p in enumerate(ps_outer):
+                z_cross_p = z_ct_outer[z_ct_outer[:,p_idx] > 0,p_idx]
```

Comment:
You can 'precompute' `z_cross_p` as a 2D array outside of the loop and filter for `>0` there (might not be easy if different rows have a different number of columns.